### PR TITLE
Added csv writer to test repo.

### PR DESCRIPTION
**Pull request by nongli**
Julien, can you make sure I'm using this right?  I've writer a csv->parquet converter and removed the parquet files from the repo.  The idea is we'd have a

csv -> parquet using mr -> read from impala -> compare with csv
and
csv -> write from impala -> read with mr -> compare with csv

### CODE REVIEW

File: parquet-mr-tests/src/test/java/parquet/compat/test/CsvWriteSupport.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			  break;
+		  case INT32:
+		      recordConsumer.addInteger(Integer.parseInt(val));
+		      break;
+		  case INT64:
+			  recordConsumer.addLong(Long.parseLong(val));
+			  break;
+		  case BINARY:
+			  recordConsumer.addBinary(stringToBinary(val));
+			  break;
```
Comment (julienledem) [inline parquet-mr-tests/src/test/java/parquet/compat/test/CsvWriteSupport.java]: default:
  throw ParquetEncodingException(...

File: parquet-mr-tests/src/test/java/parquet/compat/test/CsvWriteSupport.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	@Override
+	public void prepareForWrite(RecordConsumer r) {
+	  recordConsumer = r;		
+	}
+
+	@Override
+	public void write(List<String> values) {
+	  if (values.size() != cols.size()) {
+		  // TODO: do something useful
```
Comment (julienledem) [inline parquet-mr-tests/src/test/java/parquet/compat/test/CsvWriteSupport.java]: throw new ParquetEncodingException(...

File: parquet-mr-tests/src/test/java/parquet/compat/test/CsvWriteSupport.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		  return;
+	  }
+
+	  recordConsumer.startMessage();
+	  for (int i = 0; i < cols.size(); ++i) {		  
+		  recordConsumer.startField(cols.get(i).getPath()[0], i);
+		  String val = values.get(i);
```
Comment (julienledem) [inline parquet-mr-tests/src/test/java/parquet/compat/test/CsvWriteSupport.java]: do we want to handle nulls ?
If null you want to skip startField/endField

File: parquet-mr-tests/src/test/java/parquet/compat/test/TestReadTPCHFiles.java [hunk 4]
```diff
@@ -46,11 +54,23 @@
   private static final Log LOG = Log.getLog(TestReadTPCHFiles.class);
 
   @Test
+  public void testTpch() throws IOException {
+	File baseDir = new File("../testdata/tpch");
```
Comment (julienledem) [inline parquet-mr-tests/src/test/java/parquet/compat/test/TestReadTPCHFiles.java]: the testdata path should be named after impala of parquet-mr depending on what generated the file

File: parquet-mr-tests/src/test/java/parquet/compat/test/TestReadTPCHFiles.java [hunk 5]
```diff
@@ -46,11 +54,23 @@
...
+	final File[] csvFiles = baseDir.listFiles(new FilenameFilter() {
+	  public boolean accept(File dir, String name) {
+	    return name.endsWith(".csv");
+	  }
+	});
+	
+	for (File csvFile : csvFiles) {
+	  testMrToImpala(csvFile, "\\|");
+	}
+  }
```
Comment (julienledem) [inline parquet-mr-tests/src/test/java/parquet/compat/test/TestReadTPCHFiles.java:65]: fix indentation here

File: parquet-mr-tests/src/test/java/parquet/compat/test/TestReadTPCHFiles.java [hunk 6]
```diff
@@ -46,11 +54,23 @@
...
+  public void testTpch() throws IOException {
+	File baseDir = new File("../testdata/tpch");
+	final File[] csvFiles = baseDir.listFiles(new FilenameFilter() {
+	  public boolean accept(File dir, String name) {
+	    return name.endsWith(".csv");
+	  }
+	});
+	
+	for (File csvFile : csvFiles) {
+	  testMrToImpala(csvFile, "\\|");
```
Comment (julienledem) [inline parquet-mr-tests/src/test/java/parquet/compat/test/TestReadTPCHFiles.java]: why do we need \ here? I would expect just "|"

File: parquet-mr-tests/src/test/java/parquet/compat/test/TestReadTPCHFiles.java [hunk 7]
```diff
@@ -60,6 +80,54 @@ public boolean accept(File dir, String name) {
...
+  
+  private void testMrToImpala(File csvFile, String delimiter) throws IOException {
+    LOG.info("Converting csv file to parquet using MR: " + csvFile);
+    File parquetFile = convertToParquet(csvFile, delimiter);
+    // TODO: try to read this file from Impala	  
+  }
+  
+  static String readFile(String path) throws IOException {
+    BufferedReader reader = new BufferedReader(new FileReader(path));
+	String line = null;
```
Comment (julienledem) [inline parquet-mr-tests/src/test/java/parquet/compat/test/TestReadTPCHFiles.java]: indentation

File: parquet-mr-tests/src/test/java/parquet/compat/test/TestReadTPCHFiles.java (hunk 4 above)
Comment (nongli) [inline parquet-mr-tests/src/test/java/parquet/compat/test/TestReadTPCHFiles.java]: The test file is called blah.writeFromJava.parquet

**Comment by julienledem** [comment]
I'm looking into it, order.csv and schema do not match

**Comment by julienledem** [comment]
How do you plan to hook in read/write from Impala?

**Comment by julienledem** [comment]
Hi Nong,
Did you not plan to merge this branch?

**Pull request closed without merging**